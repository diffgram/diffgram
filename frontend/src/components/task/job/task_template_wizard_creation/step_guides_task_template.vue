<template>
  <v-container fluid data-cy="task-template-guide-step">
    <div class="d-flex mb-8 justify-space-between">
      <h1 data-cy="task-template-guide-step-title" class="font-weight-medium text--primary mr-4">
        Guides Setup:
      </h1>
      <standard_button
        tooltip_message="Create Guides"
        @click="open_guides"
        button_color="primary"
        icon="mdi-plus"
        button_message="Create Guides"
        color="white">
      </standard_button>
    </div>

    <v_error_multiple :error="error"></v_error_multiple>
    <p class="text--primary">
      Guides are used to explain annotators the requirements for correctly annotating your files.
      This is an optional step.
    </p>


    <v-container fluid>
      <h2 class="ma-0" data-cy="task-template-guide-step-subtitle">Set Your Guide (Optional): </h2>
      <guide_selector
        data-cy="guide-selector"
        :project_string_id="project_string_id"
        @change="on_change_guide"
        ref="guide_selector"
      >

      </guide_selector>
    </v-container>

    <wizard_navigation
      @next="on_next_button_click"
      :disabled_next="loading"
      :loading_next="loading_steps"
      @back="$emit('previous_step')"
      :skip_visible="false"
    >
    </wizard_navigation>

    <v-dialog v-model="dialog_open"
              :close-on-content-click="false"
              :nudge-width="200"
              offset-x>


      <v_guide_new_or_edit
        :project_string_id="project_string_id"
        :mode="'new'"
        @guide_new_success="on_guide_created">
      </v_guide_new_or_edit>

    </v-dialog>
  </v-container>

</template>

<script lang="ts">

  import axios from '../../../../services/customInstance';
  import guide_selector from '../../guide/guide_selector'

  import Vue from "vue";

  export default Vue.extend({
      name: 'step_guides_task_template',
      props: [
        'project_string_id',
        'job',
        'loading_steps'
      ],

      components: {
        guide_selector
      },

      data() {
        return {
          error: {},
          show_credentials: false,
          dialog_open: false,
          loading: false,
        }
      },
      created() {

      },

      methods: {
        on_change_guide: function (guide) {
          this.job.guide = guide;
          this.attach_selected(guide.id, 'default', 'update')
        },
        on_next_button_click: function () {
          this.$emit('next_step');
        },
        open_guides: function () {
          this.dialog_open = true;
        },
        on_guide_created: function (guide) {
          this.$refs.guide_selector.guide_list_api();
        },
        attach_selected: async function (guide_id, attach_kind, update_or_remove) {


          this.loading = true
          this.show_success_attach = false
          this.error = {}
          try {
            const response = await axios.post('/api/v1/guide/attach/job',
              {
                'update_or_remove': update_or_remove,
                'kind': attach_kind,
                'job_id': this.job.id,
                'guide_id': guide_id

              })

          } catch (e) {
            this.error = e.response.data.log.error
            console.error(e)
          }
          finally {
            this.loading = false;
          }

        },
      }
    }
  ) </script>
