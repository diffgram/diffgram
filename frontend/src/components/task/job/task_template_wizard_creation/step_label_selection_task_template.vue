<template>
  <v-container fluid data-cy="task-template-labels-step">
    <div class="d-flex mb-8 justify-space-between">
      <h1 data-cy="step-labels-title" class="font-weight-medium text--primary mr-4">
        What labels should be visible?
      </h1>
      <tooltip_button
        data-cy="manage-labels-button"
        tooltip_message="Quick Edit Project Level Schema"
        @click="open_labels_dialog"
        button_color="primary"
        icon="mdi-format-paint"
        button_message="Manage Labels & Attributes"
        color="white">
      </tooltip_button>
    </div>

    <v_error_multiple :error="error"></v_error_multiple>
    <p class="text--primary">
      Select The Labels to Use on This Task Template. All are selected by default.
    </p>
    <label_select_only
      :project_string_id="project_string_id"
      label_prompt="Schema Selected For Tasks"
      :mode=" 'multiple' "
      datacy="label-select"
      @label_file="on_change_label_file($event)"
      :load_selected_id_list="job.label_file_list"
      :select_all_at_load="true"
      ref="label_select"

    >
    </label_select_only>

    <wizard_navigation
      @next="on_next_button_click"
      :loading_next="loading_steps"
      :disabled_next="loading_steps"
      @back="$emit('previous_step')"
      :skip_visible="false"
    >
    </wizard_navigation>

    <label_manager_dialog @label_created="on_label_created"
                          :project_string_id="project_string_id"
                          ref="label_manager_dialog">

    </label_manager_dialog>
  </v-container>

</template>

<script lang="ts">

  import axios from 'axios';
  import label_select_only from '../../../label/label_select_only'
  import label_manager_dialog from '../../../label/label_manager_dialog'

  import Vue from "vue";

  export default Vue.extend({
      name: 'step_label_task_template',
      props: [
        'project_string_id',
        'job',
        'loading_steps',
      ],

      components: {
        label_select_only,
        label_manager_dialog
      },

      data() {
        return {
          error: {},
          request_refresh_labels: new Date(),
        }
      },
      created() {

      },

      computed: {

      },
      methods: {
        verify_labels: function(){
          if(!this.$props.job.label_file_list || this.$props.job.label_file_list.length === 0){
            this.error = {
              name: 'At least 1 label should be assigned to the task template.'
            }
            return false
          }
          return true
        },
        on_next_button_click: function(){
          this.error = {};
          let labels_ok = this.verify_labels();
          if(labels_ok){
            this.$emit('next_step');
          }
        },
        on_change_label_file: function(label_file_list){
          this.job.label_file_list = label_file_list;
        },
        on_label_created: function(){
          this.$refs.label_select.refresh_label_list_from_project();
        },
        open_labels_dialog: function () {
          this.$refs.label_manager_dialog.open()
        },
      }
    }
  ) </script>
