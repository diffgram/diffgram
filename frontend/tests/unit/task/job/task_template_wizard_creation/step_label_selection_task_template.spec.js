import {shallowMount, createLocalVue} from "@vue/test-utils";
import Vuex from 'vuex'
import axios from 'axios'
import step_label_selection_task_template
  from "../../../../../src/components/task/job/task_template_wizard_creation/step_label_selection_task_template";
import VueRouter from 'vue-router'
import step_guides_task_template
  from "../../../../../src/components/task/job/task_template_wizard_creation/step_guides_task_template";
const localVue = createLocalVue();
localVue.use(Vuex)
localVue.use(VueRouter)
const router = new VueRouter()

let job = {
  name: 'test',
  label_mode: 'closed_all_available',

  loading: false,
  passes_per_file: 1,
  share_object: {
    // TODO this may fail for org jobs? double check this.
    'text': String,
    'type': 'project'
  },
  share: 'project',
  instance_type: 'box', //"box" or "polygon" or... "text"...
  permission: 'all_secure_users',
  field: 'Other',
  category: 'visual',
  attached_directories_dict: {attached_directories_list: []},
  type: 'Normal',
  connector_data: {},
  // default to no review while improving review system
  review_by_human_freqeuncy: 'No review', //'every_3rd_pass'
  td_api_trainer_basic_training: false,
  file_handling: "use_existing",
  interface_connection: undefined,
  member_list_ids: ["all"]
}
describe("step_label_selection_task_template.vue", () => {
  let actions;
  let store;

  beforeEach(() => {
    actions = {
      actionClick: jest.fn(),
      actionInput: jest.fn()
    }
    store = new Vuex.Store({
      actions
    })
  });


  it("Renders component correctly.", () => {
    const wrapper = shallowMount(step_label_selection_task_template, {
      store,
      localVue,
      propsData: {
        job: job
      }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("Sets error data when calling verify_labels()", () => {
    const wrapper = shallowMount(step_label_selection_task_template, {
      store,
      localVue,
      propsData: {
        job: job
      }
    });
    wrapper.vm.verify_labels()
    expect(wrapper.vm.error).toMatchObject({
      name: 'At least 1 label should be assigned to the task template.'
    })
  });


  it("It sets label file list when calling on_change_label_file()", () => {
    const wrapper = shallowMount(step_label_selection_task_template, {
      store,
      localVue,
      propsData: {
        job: job
      }
    });
    wrapper.vm.on_change_label_file('test')
    expect(wrapper.vm.job.label_file_list).toEqual('test');
  });



});
