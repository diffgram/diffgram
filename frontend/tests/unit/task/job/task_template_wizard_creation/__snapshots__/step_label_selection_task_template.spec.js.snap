// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`step_label_selection_task_template.vue Renders component correctly. 1`] = `
<v-container fluid="" data-cy="task-template-labels-step">
  <div class="d-flex mb-8 justify-space-between">
    <h1 data-cy="step-labels-title" class="font-weight-medium text--primary mr-4">
      What labels should be visible?
    </h1>
    <tooltip_button data-cy="manage-labels-button" tooltip_message="Quick Edit Project Level Schema" button_color="primary" icon="mdi-format-paint" button_message="Manage Labels &amp; Attributes" color="white"></tooltip_button>
  </div>
  <v_error_multiple error="[object Object]"></v_error_multiple>
  <p class="text--primary">
    Select The Labels to Use on This Task Template. All are selected by default.
  </p>
  <label_select_only-stub mode="multiple" datacy="label-select" select_all_at_load="true" show_select_all="true" label_prompt="Schema Selected For Tasks"></label_select_only-stub>
  <wizard_navigation></wizard_navigation>
  <label_manager_dialog-stub></label_manager_dialog-stub>
</v-container>
`;
